### PR TITLE
Allow secure cookies for localhost

### LINKED ISSUE #459: Allow secure cookies for localhost
[Firefox 75](https://bugzilla.mozilla.org/show_bug.cgi?id=1618113) and [Chrome 89](https://bugs.chromium.org/p/chromium/issues/detail?id=1056543) now allow "Secure" cookies to be sent over plain http when set by localhost.

This is fully spec compliant, see both linked issues for details.

Should we consider a similar change for `http-client`?

It would require a little additional logic [here](https://github.com/snoyberg/http-client/blob/0a4056efb014766a3d6f099d0878c58d07144251/http-client/Network/HTTP/Client/Cookies.hs#L146).

The benefit is, in a lot of cases, you'd be able to just `{ setCookieSecure = True }` like one probably should, without having to disable it for testing and development.

This also affects downstream libraries like `wreq`.

### PR DESCRIPTION
**Pull request by mulderr**
Closes #459

Up for review. Still testing.

### CODE REVIEW

File: http-client/Network/HTTP/Client/Cookies.hs [hunk 1]
```diff
@@ -111,6 +114,37 @@ rejectPublicSuffixes = True
...
+  | isLoopbackAddr6 = True    -- step 4, part 2
+  | isLoopbackHostname = True -- step 5
+  | otherwise = False
+  where isLoopbackHostname =
+               host == "localhost"
+            || host == "localhost."
+            || BS.isSuffixOf ".localhost" host
+            || BS.isSuffixOf ".localhost." host
+        isLoopbackAddr4 =
+          fmap (head . IP.fromIPv4) (readMaybe (S8.unpack host)) == Just 127
```
Comment (snoyberg) [inline http-client/Network/HTTP/Client/Cookies.hs:138]: I'm uncomfortable about usage of a partial function here, `take 1` would be better

**Review by snoyberg — CHANGES_REQUESTED** [review]

**Comment by mulderr** [comment]
I've noticed `Read` instances for [`IPv4`](http://hackage.haskell.org/package/iproute-1.7.11/docs/src/Data.IP.Addr.html#ip4) and [`IPv6`](http://hackage.haskell.org/package/iproute-1.7.11/docs/src/Data.IP.Addr.html#ip6) skip leading whitespace. Not sure if this creates any real issues but just to be safe I'd rather refactor to use the parsers directly and be as strict as possible.

**Comment by mulderr** [comment]
On the other hand it's already parsed by network-uri and it's an internal function therefore I guess it's fine. Let me now if you'd prefer a stricter version.

**Comment by snoyberg** [comment]
I'm OK with this as-is. Have you finished your testing? Is this ready for final review?

**Comment by mulderr** [comment]
Yep! Looks good but I'm of course happy to fix any remaining issues.

**Comment by snoyberg** [comment]
Only other thing is some bookkeeping. Please:

* Perform a minor version bump in the .cabal file
* Add a ChangeLog entry

**Review by snoyberg — APPROVED** [review]
Thank you!

**Pull request merged**